app/controllers/personal_detail_controller.rb

```diff
@@ -11,7 +13,9 @@ def update
     @step = step_class.new(step_params)
 
     if @step.valid?
-      current_app.update!(step_params)
+      app = current_or_new_snap_application
+      app.update!(step_params)
+      set_current_snap_application(app)
```

Comment:
We only need to set it once (for this controller) - equivalent of "logging in". Do you still think we should create a class for this?